packages/devextreme/js/__internal/scheduler/appointment_popup/m_customize_form_items.ts

```diff
@@ -0,0 +1,115 @@
+import type {
+  FormItemComponent,
+  GroupItem,
+  Item as FormItem,
+  TabbedItem,
+} from '@js/ui/form';
+
+type ConfigItem = string | FormItem;
+
+const isGroupItem = (item: FormItem): item is GroupItem => 'items' in item && !('tabs' in item);
+const isTabbedItem = (item: FormItem): item is TabbedItem => 'tabs' in item;
+
+const BASE_SIMPLE_ITEM: FormItem = {
+  itemType: 'simple',
+  editorType: 'dxTextBox' as FormItemComponent,
+};
+
+const createFormItemFromConfig = (configItem: ConfigItem): FormItem => (
+  typeof configItem === 'string'
+    ? { ...BASE_SIMPLE_ITEM, name: configItem, dataField: configItem }
+    : { ...configItem }
+);
+
+const getChildren = (item: FormItem): FormItem[] => [
+  ...isGroupItem(item) ? item.items ?? [] : [],
+  ...isTabbedItem(item) ? item.tabs?.flatMap((tab) => tab.items ?? []) ?? [] : [],
+];
+
+const buildFormItemsMap = (
+  items: FormItem[],
+  map: Map<string, FormItem> = new Map(),
+): Map<string, FormItem> => items.reduce(
+  (accumulator, item) => {
+    if (item.name) {
+      accumulator.set(item.name, { ...item });
+    }
+    return buildFormItemsMap(getChildren(item), accumulator);
+  },
+  map,
+);
+
+const removeItemFromGroups = (
+  itemName: string,
+  itemsMap: Map<string, FormItem>,
+): void => {
+  Array.from(itemsMap.values()).forEach((group) => {
+    if (isGroupItem(group) && group.items) {
+      group.items = [...group.items.filter((item) => item.name !== itemName)];
+    }
+  });
+};
+
+const processGroupItems = (newItem: FormItem, itemsMap: Map<string, FormItem>): FormItem => (
+  isGroupItem(newItem) && newItem.items
```

Comment:
Probably we can replace this line by
```js
isGroupItem(newItem)
```

because isGroupItem() already checks that newItems.items is defined